## Task 6 — Plan Revenue by Country

### Business Question

Which subscription plan generates the highest monthly revenue in each country?

### Analysis

Group customers by country and subscription plan, calculate the total monthly revenue for each combination, sort the results by total revenue in descending order, and identify the highest-revenue plan for each country.

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
revenue_by_plan_and_country = starlink.groupby(["country","plan_name"]).agg(
    total_revenue = ("monthly_fee","sum")
).sort_values(
    "total_revenue",
    ascending = False
).groupby("country").head(1)
print(revenue_by_plan_and_country)

                            total_revenue
country        plan_name                 
Australia      Residential        77280.0
Germany        Residential        75600.0
Canada         Residential        73920.0
Ukraine        Residential        73320.0
Spain          Residential        72480.0
Brazil         Residential        70440.0
France         Residential        69000.0
United States  Residential        68280.0
Japan          Residential        67920.0
United Kingdom Residential        66840.0


### Result

The Residential plan generates the highest total monthly revenue in all analyzed countries. Australia has the highest Residential revenue at 77,280, followed by Germany at 75,600 and Canada at 73,920.

### Business Insight

The Residential plan generates the highest total monthly revenue in every country. This indicates that Residential is the main revenue-generating subscription plan across all analyzed markets. However, the analysis does not show whether this is caused by a higher number of Residential customers, a higher subscription fee, or both.